### PR TITLE
Use https in documentation

### PR DESCRIPTION
**Pull request by mcornella**
As [Paul Irish put it](https://twitter.com/paul_irish/status/588502455530311680):

> The protocol relative URL is now an anti-pattern. If the asset is available on HTTPS, always request it with that. https://www.paulirish.com/2010/the-protocol-relative-url/

This ensures that forms are always submitted using HTTPS, even on websites delivered by HTTP.

-----
**Fixes partially #85.**

**Changes proposed in this pull request:**

* Drop the use of the protocol-relative URL (`//host/...`) in favor of always using HTTPS (`https://host/...`).
* Change the home page URL to HTTPS.
* Change the `API_ROOT` config variable: now it must begin with `https://`.

**Screenshots and GIFs**

See website screenshots over at https://github.com/formspree/formspree/issues/85#issuecomment-197026612.

icanhazhttps

**Deploy notes**

The `API_ROOT` environment variable should be changed to start with `https://`. As I said before, I can make the commit so that it doesn't require changing it, i.e. by hardcoding `https://` on forms that use `config.API_ROOT`.

### CODE REVIEW

**Comment by rohitdatta** [comment]
This is an interesting question. Really, putting `https` in front of everything is a pretty bad idea, since this code should be able to be deployed on servers that don't support HTTPS (such as localhost). I think the best solution would be just to change the API_ROOT from `//formspree.io` to `https://formspree.io`, but we need to make sure that that doesn't mess up anything else. I don't believe this will, since it doesn't appear to be used anywhere else at a first glance, but we'll need to look further into this.

**Comment by fiatjaf** [comment]
I agree with @rohitdatta.

**Comment by fiatjaf** [comment]
Besides changing the `API_ROOT` config var (or perhaps adding a new `PROTOCOL` config var), I'm against merging this. It doesn't do anything good for anyone and it hurts people that may be running their Formspree instances without HTTP.

**Comment by rohitdatta** [comment]
Yeah from what I can tell, the only change necessary for this would be to change the API_ROOT to `https://formspree.io`, but I still can't tell if that will break anything. I don't think a `PROTOCOL` variable would be necessary. If changing the API_ROOT breaks anything I don't know what the solution is, but I wouldn't add a layer of complexity with a protocol variable.

**Comment by colevscode** [comment]
If we bake the protocol into the API_ROOT it would still support people deploying on localhost, they would just set the API_ROOT to http://localhost:8000/ or whatever.

This PR is pretty close. I just suggest scratching "https://127.0.0.1" because that's pretty silly. The rest of the changes are OK I think, as long as we change the API_ROOT.

Other than 1) testing that our ssl infrastructure is fast enough for https (https://istlsfastyet.com/), 2) merging this PR and 3) updating API_ROOT, what else do we need to do to switch over to https?

File: formspree/settings.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 CONTACT_EMAIL = os.getenv('CONTACT_EMAIL') or 'team@example.com'
 DEFAULT_SENDER = os.getenv('DEFAULT_SENDER') or 'Forms Team <submissions@example.com>'
 ACCOUNT_SENDER = os.getenv('ACCOUNT_SENDER') or DEFAULT_SENDER
-API_ROOT = os.getenv('API_ROOT') or '//example.com'
+API_ROOT = os.getenv('API_ROOT') or 'https://example.com'
```
Comment (fiatjaf) [inline formspree/settings.py]: I think this is unnecessary.

File: README.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ Once your environment is setup, create a postgresql database, clone the source a
 
 then create a `.env` file with your configuration like the following:
 
-    API_ROOT='127.0.0.1:5000'
+    API_ROOT='https://127.0.0.1:5000'
```
Comment (fiatjaf) [inline README.md]: This is wrong.

**Comment by rohitdatta** [comment]
Well so if we change API_ROOT to `https://formspree.io` (it's `//formspree.io` currently) I don't think the documentation except the README needs to get updated since it pulls the submission values in the documentation from the config vars. I haven't found any place in the code itself which uses API_ROOT, it seems like it's just a variable for the documentation, so if our SSL infrastructure is fast enough for HTTPS, just changing the API_ROOT should be enough and the documentation should update accordingly?

**Comment by mcornella** [comment]
Ok, I got rid of the modifications to the `API_ROOT` variable, but I left the modifications to the `template/static_pages/` html action URLs because these seemed to be inconsistent. See the latest commit at d37ac5032e36ec6a7cd87e3b620e4aae7fe7079c.

**Comment by fiatjaf** [comment]
Are we going to merge this? I have this [related PR](https://github.com/formspree/formspree/pull/89) that would go well after this one.

**Comment by rohitdatta** [comment]
I'm fine with merging this. We can try changing the API_ROOT on Heroku and test out all the scenarios. I didn't find any instance in the code so I'm fairly certain we're OK with switching it from protocol relative to HTTPS.

**Comment by mcornella** [comment]
The website is already updated I see:

web1
web2
web3

Thanks! :heart_eyes:

**Pull request merged**